app/src/main/java/com/duckduckgo/app/bookmarks/ui/BookmarksViewModel.kt

```diff
@@ -142,19 +138,89 @@ class BookmarksViewModel @Inject constructor(
                 }
             }
         }
-        delete(savedSite)
+        hide(savedSite)
     }
 
-    private fun delete(savedSite: SavedSite) {
+    private fun hide(savedSite: SavedSite) {
+        // we don't delete, just remove from the ui list
+        when (savedSite) {
+            is Bookmark -> {
+                val bookmarks = viewState.value?.bookmarks!!.toMutableList()
+                val index = bookmarks.indexOf(savedSite)
+                bookmarks[index] = savedSite.copy(deleted = "1")
+                val bookmarkFolders = viewState.value?.bookmarkFolders!!
+
+                val favourites = viewState.value?.favorites!!.toMutableList()
+                val favourite = favourites.find { it.id == savedSite.id }!!
+                val favouritesIndex = favourites.indexOf(favourite)
+                favourites[favouritesIndex] = favourite.copy(deleted = "1")
+
+                viewState.value = viewState.value?.copy(
+                    favorites = favourites,
+                    bookmarks = bookmarks,
+                    enableSearch = bookmarks.size + bookmarkFolders.size >= MIN_ITEMS_FOR_SEARCH,
+                )
+            }
+
+            is Favorite -> {
+                val favourites = viewState.value?.favorites!!.toMutableList()
+                val index = favourites.indexOf(savedSite)
+                favourites[index] = savedSite.copy(deleted = "1")
+
+                viewState.value = viewState.value?.copy(
+                    favorites = favourites,
+                )
+            }
+        }
+    }
+
+    fun delete(savedSite: SavedSite) {
         viewModelScope.launch(dispatcherProvider.io() + NonCancellable) {
-            faviconManager.deletePersistedFavicon(savedSite.url)
+            if (savedSite is Bookmark) {
+                faviconManager.deletePersistedFavicon(savedSite.url)
+            }
             savedSitesRepository.delete(savedSite)
         }
     }
 
-    fun insert(savedSite: SavedSite) {
-        viewModelScope.launch(dispatcherProvider.io()) {
-            savedSitesRepository.insert(savedSite)
+    fun undoDelete(savedSite: SavedSite) {
+        // we don't insert it, only flip the deleted flag
+        when (savedSite) {
+            is Bookmark -> {
+                val bookmarks = viewState.value?.bookmarks!!.toMutableList()
```

Comment:
Posted a suggestion in `BrowserTabViewModel` to make this simpler emitting a list of hidden ids.

With that we don't need to flip deleted flag, and we all the logic will belong to the viewmodel.